src/main/java/com/atharvadholakia/password_manager/controller/CredentialController.java

```diff
@@ -34,6 +34,29 @@ public ResponseEntity<Credential> createCredential(@Valid @RequestBody Credentia
     return new ResponseEntity<>(createdCredential, HttpStatus.CREATED);
   }
 
+  @PatchMapping("/{id}")
```

Comment:
Please check when to use `patch` and when to use `put`.

Reply:
Yes, I have checked it. Patch can be used to make partial updates for e.g only updating username. PUT does not support partial updates. It will completely replace the whole data with the new one.